switchlink/switchlink_link.c

```diff
@@ -88,6 +93,51 @@ static switchlink_link_type_t get_link_type(const char* info_kind) {
   return link_type;
 }
 
+
+#if defined(ES2K_TARGET)
+/*
+ * Routine Description:
+ *    Check if the interface driver is valid for our use case
+ *
+ * Arguments:
+ *    [in] ifname - Interface name
+ *
+ * Return Values:
+ *    boolean
+ */
+bool validate_driver_name(char* ifname) {
+  struct ethtool_drvinfo drv = {0};
+  char drvname[32] = {0};
+  struct ifreq ifr = {0};
+  int fd, r = 0;
+
+  fd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (fd < 0) {
+    return false;
+  }
+
+  drv.cmd = ETHTOOL_GDRVINFO;
+  strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
+  ifr.ifr_data = (void*)&drv;
+
+  r = ioctl(fd, SIOCETHTOOL, &ifr);
+  if (r) {
+    goto end;
```

Comment:
It is always beneficial for debugging to log the error details at this point.  This applies to every instance in this function where error messages aren't logged before returning.